Makefile

```diff
@@ -97,15 +102,27 @@ $(1)-$(2): bindir
 	$(DOCKER_BUILD) 'GOARCH=amd64 GOOS=$(2) $(GO_BUILDCMD) -o $(1)-$(2) cmd/$(3)/main.go'
 ALL_BINS := $(ALL_BINS) $(1)-$(2)
 endef
+
+define E2E_OS_template
+$(1)-$(2): bindir
+	$(DOCKER_GOPATH_BUILD) 'GOPATH=$(GOPATH) GOARCH=amd64 GOOS=$(2) $(GO_TESTCMD) -o $(1)-$(2) ./test/$(3)'
+ALL_BINS := $(ALL_BINS) $(1)-$(2)
+endef
+
```

Comment:
it seems moving `E2E_OS_template` after line 112 is better in code locality. Line 112 handles `OS_template` define above.

Reply:
Sure, I will fix this right away.